client_perms/server.sh

```diff
@@ -0,0 +1,278 @@
+		cat /var/log/ganesha.log
+		echo
+		echo "+++ /etc/ganesha/ganesha.conf +++"
+		grep --with-filename -e '' /etc/ganesha/ganesha.conf
+		echo
+		echo "+++ /etc/ganesha/exports/*.conf +++"
+		grep --with-filename -e '' /etc/ganesha/exports/*.conf
+		echo
+		echo "Export ${GLUSTER_VOLUME} is not available"
+		exit 1
+	fi
+
+	echo "============Displaying Initial Configuration File============================="
+	cat /etc/ganesha/exports/export.${GLUSTER_VOLUME}.conf
+
+	#Enabling ACL for the volume if ENABLE_ACL param is set to True
+	if [ "${ENABLE_ACL}" == "True" ]
+	then
+	  sed -i s/'Disable_ACL = .*'/'Disable_ACL = false;'/g ${conf_file}
+	  cat ${conf_file}
+
+	  dbus-send --type=method_call --print-reply --system  --dest=org.ganesha.nfsd /org/ganesha/nfsd/ExportMgr  org.ganesha.nfsd.exportmgr.UpdateExport string:/etc/ganesha/exports/export.${GLUSTER_VOLUME}.conf string:"EXPORT(Export_Id = 2)"
+	fi
+
+fi
+
+if [ "$1" = "server_stage1" ]
+then
+	echo "======= SERVER STAGE 1 =========="
+	touch clientBlock.txt
+	echo -e "CLIENT{
+	\tClients = ${CLIENT};
+        \tSquash = \"no_root_squash\";
+        \tAccess_type = \"RO\";
+        \tProtocols = \"3\",\"4\";
+	}" > clientBlock.txt
+	
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	echo "=======CLIENTBLOCK========="
+	cat clientBlock.txt
+	sed -i '19e cat clientBlock.txt' ${conf_file}
+
+	echo "UPDATED EXPORT FILE"
+	cat ${conf_file}
+
+	dbus-send --type=method_call --print-reply --system  --dest=org.ganesha.nfsd /org/ganesha/nfsd/ExportMgr  org.ganesha.nfsd.exportmgr.UpdateExport string:${conf_file} string:"EXPORT(Export_Id = ${export_id})"
+
+	sleep 15
+	echo "++++++++Export Data Updated+++++++++"
+fi
+
+if [ "$1" = "server_stage2" ]
+then
+	echo "======= SERVER STAGE 2 =========="
+
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	sed -i '22s/.*/\t\tAccess_type = "RW";/' ${conf_file}
+	sed -i '23s/.*/\t\tProtocols = "3";/' ${conf_file}
+
+	echo "UPDATED EXPORT FILE"
+	cat ${conf_file}
+
+	dbus-send --type=method_call --print-reply --system  --dest=org.ganesha.nfsd /org/ganesha/nfsd/ExportMgr  org.ganesha.nfsd.exportmgr.UpdateExport string:${conf_file} string:"EXPORT(Export_Id = ${export_id})"
+
+	sleep 15
+	echo "++++++++Export Data Updated+++++++++"
+fi
+
+if [ "$1" = "server_stage3" ]
+then
+	echo "======= SERVER STAGE 3 =========="
+
+	conf_file="/etc/ganesha/exports/export."${GLUSTER_VOLUME}".conf"
+
+	#Parsing export id from volume export conf file
+	export_id=$(grep 'Export_Id' ${conf_file} | sed 's/^[[:space:]]*Export_Id.*=[[:space:]]*\([0-9]*\).*/\1/')
+
+	sed -i '23s/.*/\t\tProtocols = "4";/' ${conf_file}
```

Comment:
Same for this...

Reply:
I don't understand what the code is trying to do, but does "ganesha_conf" script shipped with ganesha2.5 help? It should add/replace/delete existing config entries. See ./src/scripts/ganeshactl/ganesha_conf.man for details.